consensus/event_buffer.go

```diff
@@ -0,0 +1,45 @@
+package consensus
+
+import (
+	"github.com/tendermint/tendermint/types"
+)
+
+const (
+	eventBufferCapacity = 1000
+)
+
+// Interface assertions
+var _ types.EventsPublisher = (*EventBuffer)(nil)
+
+// EventBuffer is a buffer of events.
+type EventBuffer struct {
```

Comment:
Note: there is already some buffering since I am using buffered channels. But EventBuffer gives you control on when to fire/process events.